design/ceph-csi-driver.md

```diff
@@ -54,6 +56,19 @@ If CSI drivers are already successfully deployed, Rook should help monitor the S
 
 ### Rook initiated CSI drivers deployment
 
-If Rook is instructed to deploy CSI drivers and other external controllers, Rook Operator should create all the driver and daemon deployments as well.
+If Rook is instructed to deploy CSI drivers and other external controllers, Rook Operator should create all the driver and daemon deployments as well. Since there are breaking changes in CSI 1.0 spec and Kubernetes 1.13, it makes sense to focus on newer CSI specs and only on latest Kubernetes releases (i.e. 1.13 and beyond).
+
+The changes to Rook include the following:
+
+- Create a new API object for CSI driver deployment. Such object should inform Rook Ceph Operator which driver (rbd or Ceph FS, or both) to start, in which namespace to run the CSI drivers, and which Ceph cluster to get configuration infomation to create Storage Class and Secret.
```

Comment:
Several questions:
- What API object? Could these be env variables in the operator.yaml? That's how we currently pass options to the operator, but there might be a better way.
- In what namespace(s) do the CSI drivers need to start? Do they start in every namespace where clients want to mount the storage?
- The admin creates the storage class. Would the operator generate the secret when it sees a new storage class?
- The operator would have to update the secret for every storage class when there is a mon failover?